app/models/background.rb

```diff
@@ -11,10 +11,30 @@ class Background < ApplicationRecord
   DEFAULT_BACKGROUND_MODEL_NAME = "default".freeze
   TAXON_SUMMARY_CHUNK_SIZE = 100
 
+  # This set of ten backgrounds was generated by SQL analysis of sample page
+  # requests for the purpose of precache_report_info. The top 1 accounts for
+  # 52% of unique users, top 5 accounts for 71%, and top 10, 83%.
+  TOP_BY_USAGE = [
```

Comment:
I disagree with this approach a static on the moment analysis.
This is not a scalable approach, and is not something we can generalize to other pages.
If we want to do something like this, then we need to at least generate this list dynamically, e.g. cron task that computes this stats and adds them to our database.

Reply:
Yes, this static list might fall out of sync with actual usage over time. There are other ways we can rank backgrounds by usage such as:

* host_genomes default_background_id
* pipeline_run background_id
* sample background_id

Do any of those make sense? 

We could also limit the backgrounds to those that have `public_access`

Reply:
`public_access` seems to be a good one! I believe the default_backgroud_id would also be included in that set  -- it's required to be public or the default view would break.

Not sure what you mean by "background_id of the sample or pipeline being cached". There is an association between backgrounds and pipeline_runs, but it works the other way around: the background is generated from a set of pipeline_runs

Reply:
Could add to that any backgrounds created by the user who uploaded the sample (maybe top 10 starting with the most recent ones?). I.e. where background.user_id == sample.user_id

Reply:
Your last paragraph seems to contain the best two alternatives. 
Agree, with setting a limit as @cdebourcy suggests, but would probably set a shorted limited.
For public samples, would make sense as long as we keep making a background public restricted (i.e. AFAIK there is no way now for a user to a make a background public).

A bit more convoluted third option, to cover team projects with multiple uploaders or where uploader != analyst: backgrounds that contain samples of the project that the samples below belong too.

Of course, the alternatives are not mutually exclusive.

Are we already measuring the effectiveness of the cache? Could be good to keep track of the backgrounds/samples for which it misses.

Reply:
Thanks for the input. I pushed some changes. See the updated PR description.